frameworks/hdfs/tests/test_sanity.py

```diff
@@ -100,64 +101,63 @@ def check_properties(xml, expect):
 
 @pytest.mark.recovery
 def test_kill_journal_node():
-    foldered_name = sdk_utils.get_foldered_name(config.SERVICE_NAME)
-    journal_ids = sdk_tasks.get_task_ids(foldered_name, "journal-0")
+    journal_task = sdk_tasks.get_service_tasks(foldered_name, "journal-0")[0]
     name_ids = sdk_tasks.get_task_ids(foldered_name, "name")
     data_ids = sdk_tasks.get_task_ids(foldered_name, "data")
 
-    sdk_cmd.kill_task_with_pattern(
-        "journalnode", sdk_hosts.system_host(foldered_name, "journal-0-node")
-    )
+    sdk_cmd.kill_task_with_pattern("journalnode", "nobody", agent_host=journal_task.host)
+
     config.expect_recovery(service_name=foldered_name)
-    sdk_tasks.check_tasks_updated(foldered_name, "journal", journal_ids)
+    sdk_tasks.check_tasks_updated(foldered_name, "journal", [journal_task.id])
     sdk_tasks.check_tasks_not_updated(foldered_name, "name", name_ids)
     sdk_tasks.check_tasks_not_updated(foldered_name, "data", data_ids)
 
 
 @pytest.mark.sanity
 @pytest.mark.recovery
 def test_kill_name_node():
-    foldered_name = sdk_utils.get_foldered_name(config.SERVICE_NAME)
-    name_ids = sdk_tasks.get_task_ids(foldered_name, "name-0")
+    name_task = sdk_tasks.get_service_tasks(foldered_name, "name-0")[0]
     journal_ids = sdk_tasks.get_task_ids(foldered_name, "journal")
     data_ids = sdk_tasks.get_task_ids(foldered_name, "data")
 
-    sdk_cmd.kill_task_with_pattern("namenode", sdk_hosts.system_host(foldered_name, "name-0-node"))
+    sdk_cmd.kill_task_with_pattern("namenode", "nobody", agent_host=name_task.host)
+
     config.expect_recovery(service_name=foldered_name)
-    sdk_tasks.check_tasks_updated(foldered_name, "name", name_ids)
+    sdk_tasks.check_tasks_updated(foldered_name, "name", [name_task.id])
     sdk_tasks.check_tasks_not_updated(foldered_name, "journal", journal_ids)
     sdk_tasks.check_tasks_not_updated(foldered_name, "data", data_ids)
 
 
 @pytest.mark.sanity
 @pytest.mark.recovery
 def test_kill_data_node():
-    foldered_name = sdk_utils.get_foldered_name(config.SERVICE_NAME)
-    data_ids = sdk_tasks.get_task_ids(foldered_name, "data-0")
+    data_task = sdk_tasks.get_service_tasks(foldered_name, "data-0")[0]
     journal_ids = sdk_tasks.get_task_ids(foldered_name, "journal")
     name_ids = sdk_tasks.get_task_ids(foldered_name, "name")
 
-    sdk_cmd.kill_task_with_pattern("datanode", sdk_hosts.system_host(foldered_name, "data-0-node"))
+    sdk_cmd.kill_task_with_pattern("datanode", "nobody", agent_host=data_task.host)
```

Comment:
Question: do we ever call this without `nobody`?

Reply:
Yes, in hello-world where it's killing tasks on the master: mesos-master (root) and zookeeper (dcos_exhibitor)